### PR TITLE
add TLS encryption

### PR DESCRIPTION
**Pull request by e-max**
Driver can use TLS connection if  CA pem, certificate and key are set in configuration.

### CODE REVIEW

File: connection.go [hunk 1]
```diff
@@ -29,6 +32,45 @@ type connection struct {
 	readTimeout time.Duration
 }
 
+func newTLSConnection(address string, ca, cert, key []byte, timeout, readTimeout time.Duration) (*connection, error) {
+	roots := x509.NewCertPool()
+	ok := roots.AppendCertsFromPEM(ca)
+	if !ok {
+		panic("failed to parse root certificate")
```
Comment (hashmap) [inline connection.go]: why panic if we return error from the function? also below there is `log.panic`, here `panic`, should we unify?

File: connection.go [hunk 2]
```diff
@@ -29,6 +32,45 @@ type connection struct {
...
+		Certificates: []tls.Certificate{certificate},
+		RootCAs:      roots,
+	}
+
+	dialer := net.Dialer{
+		Timeout:   timeout,
+		KeepAlive: 30 * time.Second,
+	}
+	conn, err := tls.DialWithDialer(&dialer, "tcp", address, conf)
+	//conn, err := dialer.Dial("tcp", address)
```
Comment (hashmap) [inline connection.go]: remove?

**Comment by e-max** [comment]
Added test for TLS connections

**Comment by e-max** [comment]
I need to rebase this PR on master, so I'll recreate it

**Pull request closed without merging**